src/client/pkg/grpcutil/server.go

```diff
@@ -47,9 +78,28 @@ func Serve(
 				PermitWithoutStream: true,
 			}),
 		}
+		if server.PublicPortTLSAllowed {
+			// Validate environment
+			certPath := path.Join(TLSVolumePath, TLSCertFile)
+			keyPath := path.Join(TLSVolumePath, TLSKeyFile)
+			_, certPathStatErr := os.Stat(certPath)
+			_, keyPathStatErr := os.Stat(keyPath)
+			if certPathStatErr != nil {
+				log.Warnf("TLS enabled but could not stat private key at %s: %e", certPath, certPathStatErr)
+			} else if keyPathStatErr != nil {
+				log.Warnf("TLS enabled but could not stat private key at %s: %e", certPath, keyPathStatErr)
```

Comment:
This error handling seems a bit weird because it only prints out the warning for at most one of the files, even if there's errors with both. Could we just have 3 ifs here, one for each error and then one if both errors are nil?

Reply:
Done